pkg/apis/eventbus/v1alpha1/jetstream_eventbus.go

```diff
@@ -0,0 +1,96 @@
+	// Optional: Defaults to empty.  See type description for default values of each field.
+	// +optional
+	SecurityContext *corev1.PodSecurityContext `json:"securityContext,omitempty" protobuf:"bytes,9,opt,name=securityContext"`
+	// ImagePullSecrets is an optional list of references to secrets in the same namespace to use for pulling any of the images used by this PodSpec.
+	// If specified, these secrets will be passed to individual puller implementations for them to use. For example,
+	// in the case of docker, only DockerConfig type secrets are honored.
+	// More info: https://kubernetes.io/docs/concepts/containers/images#specifying-imagepullsecrets-on-a-pod
+	// +optional
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,10,rep,name=imagePullSecrets"`
+	// If specified, indicates the Redis pod's priority. "system-node-critical"
+	// and "system-cluster-critical" are two special keywords which indicate the
+	// highest priorities with the former being the highest priority. Any other
+	// name must be defined by creating a PriorityClass object with that name.
+	// If not specified, the pod priority will be default or zero if there is no
+	// default.
+	// More info: https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/
+	// +optional
+	PriorityClassName string `json:"priorityClassName,omitempty" protobuf:"bytes,11,opt,name=priorityClassName"`
+	// The priority value. Various system components use this field to find the
+	// priority of the Redis pod. When Priority Admission Controller is enabled,
+	// it prevents users from setting this field. The admission controller populates
+	// this field from PriorityClassName.
+	// The higher the value, the higher the priority.
+	// More info: https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/
+	// +optional
+	Priority *int32 `json:"priority,omitempty" protobuf:"bytes,12,opt,name=priority"`
+	// The pod's scheduling constraints
+	// More info: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty" protobuf:"bytes,13,opt,name=affinity"`
+	// ServiceAccountName to apply to the StatefulSet
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName,omitempty" protobuf:"bytes,14,opt,name=serviceAccountName"`
+	// JetStream configuration, if not specified, global settings in controller-config will be used.
+	// See https://docs.nats.io/running-a-nats-service/configuration#jetstream.
+	// Only configure "max_memory_store" or "max_file_store", do not set "store_dir" as it has been hardcoded.
+	// +optional
+	Settings *string `json:"settings,omitempty" protobuf:"bytes,15,opt,name=settings"`
+}
+
+func (j JetStreamBus) GetReplicas() int {
+	if j.Replicas == nil {
+		return 3
+	}
+	if *j.Replicas < 3 {
```

Comment:
oh, interesting. We don't let them have less than 3. Ideally maybe we'd warn them but not critical.

Reply:
Cluster setting has a different configuration from single node setting, it's not worth supporting that for us.

For a cluster setting, it needs at least 2 nodes to make it work. JetStream uses a RAFT algorithm for clustering, which does not make sense to use an even number. So a reasonable cluster node number is 3 or 5.

```
In order to ensure data consistency across complete restarts, a quorum of servers is required. A quorum is ½ cluster size + 1. This is the minimum number of nodes to ensure at least one node has the most recent data and state after a catastrophic failure. So for a cluster size of 3, you’ll need at least two JetStream enabled NATS servers available to store new messages. For a cluster size of 5, you’ll need at least 3 NATS servers, and so forth.
```

https://docs.nats.io/running-a-nats-service/configuration/clustering/jetstream_clustering#the-quorum

Reply:
Okay. Just wondering if we might want to log something so it's not a silent change. But I'm good with whatever - probably rare that somebody would set replicas to 1 or 2.

Reply:
Adding a log sounds good.

Reply:
It seems to be awkward to add a log in the model, skip it for now.